clang/test/SemaSYCL/__failed_assertion_fail.cpp

```diff
@@ -0,0 +1,17 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -verify -fsyntax-only %s
+// UNSUPPORTED: system-windows
+// This test checks that an undefined "__failed_assertion" without SYCL_EXTERNAL which is not declared
+// in a system header will lead to SYCL sema check failure.
+
+#include "sycl.hpp"
+
+void __failed_assertion();
+// expected-note@-1 {{'__failed_assertion' declared here}}
+
+SYCL_EXTERNAL
+void call_failed_assertion() {
+  // expected-note@-1 {{called by 'call_failed_assertion'}}
+  __failed_assertion();
+  // expected-error@-1 {{SYCL kernel cannot call an undefined function without SYCL_EXTERNAL attribute}}
+}
+
```

Comment:
Let's put both tests into a single file instead of having a pass/fail pair. Something like this should work:
```
// RUN: %clang_cc1 -fsycl-is-device -sycl-std=2020 -verify=user -DUSER -fsyntax-only %s
// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2020 -verify=sys -fsyntax-only %s
// UNSUPPORTED: system-windows

#ifdef USER
void __failed_assertion(); // user-note {{'__failed_assertion' declared here}}
#else
#include <dummy_failed_assert>
#endif
#include "sycl.hpp"

// sys-no-diagnostics

SYCL_EXTERNAL void call_failed_assertion() {   // user-note {{called by 'call_failed_assertion'}}
  __failed_assertion();  // user-error {{SYCL kernel cannot call an undefined function without SYCL_EXTERNAL attribute}}
}
```